### ISSUE TITLE
[RAPT] Problem when install binary packages into an existing image

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#454]
**Reported by valleegr on 5 Feb 2008 21:28 UTC**
If i try to install the binary package opk-c3-client into the partition1 image with the command ''/usr/bin/rapt --installroot /var/lib/systemimager/images/partition1/ install opkg-c3-client'', i have the following error:

```
# /usr/bin/rapt --installroot /var/lib/systemimager/images/partition1/ install opkg-c3-client
Executing: /home/gvh/src/svn/oscar/scripts/distro-query --image /var/lib/systemimager/images/partition1/
Repositories detected:
        http://bear.csm.ornl.gov/debian/+etch+main
        file:/tftpboot/oscar/debian-4-x86_64,file:/tftpboot/oscar/common-debs,http://oscar.gforge.inria.fr/debian/+stable+main+oscar
WARNING: repository name doesn't match any valid architecture
Executing: chroot /var/lib/systemimager/images/partition1/ apt-get update
Ign http://127.0.0.1 stable Release.gpg
Ign http://127.0.0.1 stable Release
Err http://127.0.0.1 stable/main Packages
  404 Not Found
Get:1 http://bear.csm.ornl.gov etch Release.gpg [http://127.0.0.1 stable/oscar Packages
  404 Not Found
Get:2 http://bear.csm.ornl.gov etch Release [58.2kB](378B]
Err)
Get:3 http://bear.csm.ornl.gov etch/main Packages [5538kB]
Fetched 5597kB in 0s (6052kB/s)
Failed to fetch http://127.0.0.1/repo/tftpboot/oscar/debian-4-x86_64,file:/tftpboot/oscar/common-debs,http://oscar.gforge.inria.fr/debian/dists/stable/main/binary-amd64/Packages.gz  404 Not Found
Failed to fetch http://127.0.0.1/repo/tftpboot/oscar/debian-4-x86_64,file:/tftpboot/oscar/common-debs,http://oscar.gforge.inria.fr/debian/dists/stable/oscar/binary-amd64/Packages.gz  404 Not Found
Reading package lists... Done
E: Some index files failed to download, they have been ignored, or old ones used instead.
Error occured while running "chroot /var/lib/systemimager/images/partition1/ apt-get update" (25600) at /usr/bin/rapt line 213.
```

It seems that the list of repos is not correctly handled.

**Comment by dikim33** [comment]
**Modified by valleegr on 5 Feb 2008 21:35 UTC**

**Comment by dikim33** [comment]
**Comment by valleegr on 5 Feb 2008 22:48 UTC**
(In [6920]) This check-in fixes #455 - fix a bug when handling multiple repositories.
